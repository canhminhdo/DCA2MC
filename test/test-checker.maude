load ../debug.maude
load ../full-maude-lite.maude
load ../specs/tas.maude
load ../utils.maude
load ../database.maude
load ../checker.maude

red in DCA2MC-CHECKER : tryEvalOne(
    upModule('TAS-CHECK, false),
    buildTableau('_R_['False.FalseFormula, '_\/_['~_['inWs1.Prop], '_U_['True.TrueFormula,'inCs1.Prop]]]),
    'init.Config,
    getInitNode(buildTableau('_R_['False.FalseFormula, '_\/_['~_['inWs1.Prop], '_U_['True.TrueFormula,'inCs1.Prop]]]))
) .

red layerCheck(
    upModule('TAS-CHECK, false),
    buildTableau('_R_['False.FalseFormula, '_\/_['~_['inWs1.Prop], '_U_['True.TrueFormula,'inCs1.Prop]]]),
    'VAR:Config,
    ({('init.Config) : getInitNode(buildTableau('_R_['False.FalseFormula, '_\/_['~_['inWs1.Prop], '_U_['True.TrueFormula,'inCs1.Prop]]]))}),
    2
) .