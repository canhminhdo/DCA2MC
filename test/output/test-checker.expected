==========================================
reduce in DCA2MC-CHECKER : tryEvalOne(upModule('TAS-CHECK, false),
    buildTableau('_R_['False.FalseFormula, '_\/_['~_['inWs1.Prop], '_U_[
    'True.TrueFormula, 'inCs1.Prop]]]), 'init.Config, getInitNode(buildTableau(
    '_R_['False.FalseFormula, '_\/_['~_['inWs1.Prop], '_U_['True.TrueFormula,
    'inCs1.Prop]]]))) .
result Result4TryEval: < (unknown).ResultType | {'_R_['False.FalseFormula,
    '_\/_['~_['inWs1.Prop], '_U_['True.TrueFormula, 'inCs1.Prop]]]} >
==========================================
reduce in DCA2MC-CHECKER : layerCheck(upModule('TAS-CHECK, false),
    buildTableau('_R_['False.FalseFormula, '_\/_['~_['inWs1.Prop], '_U_[
    'True.TrueFormula, 'inCs1.Prop]]]), 'VAR:Config, {'init.Config :
    getInitNode(buildTableau('_R_['False.FalseFormula, '_\/_['~_['inWs1.Prop],
    '_U_['True.TrueFormula, 'inCs1.Prop]]]))}, 2) .
result Result4Checking: < (unknown).ResultType | {'`{_`}['__['locked:_[
    'false.Bool], 'cnt:_['s_^2['0.Zero]], 'pc`[_`]:_['p1.Pid, 'ws.Loc],
    'pc`[_`]:_['p2.Pid, 'ws.Loc]]] : {'_R_['False.FalseFormula, '_\/_['~_[
    'inWs1.Prop], '_U_['True.TrueFormula, 'inCs1.Prop]]]}}, {'`{_`}['__[
    'locked:_['false.Bool], 'cnt:_['s_^2['0.Zero]], 'pc`[_`]:_['p1.Pid,
    'ws.Loc], 'pc`[_`]:_['p2.Pid, 'ws.Loc]]] : {('_R_['False.FalseFormula,
    '_\/_['~_['inWs1.Prop], '_U_['True.TrueFormula, 'inCs1.Prop]]])
| ('_U_['True.TrueFormula, 'inCs1.Prop])}}, {'`{_`}['__['locked:_['true.Bool],
    'cnt:_['s_^2['0.Zero]], 'pc`[_`]:_['p1.Pid, 'cs.Loc], 'pc`[_`]:_['p2.Pid,
    'ss.Loc]]] : {('_R_['False.FalseFormula, '_\/_['~_['inWs1.Prop], '_U_[
    'True.TrueFormula, 'inCs1.Prop]]])
| ('_U_['True.TrueFormula, 'inCs1.Prop])}}, {'`{_`}['__['locked:_['true.Bool],
    'cnt:_['s_^2['0.Zero]], 'pc`[_`]:_['p1.Pid, 'ss.Loc], 'pc`[_`]:_['p2.Pid,
    'cs.Loc]]] : {'_R_['False.FalseFormula, '_\/_['~_['inWs1.Prop], '_U_[
    'True.TrueFormula, 'inCs1.Prop]]]}} >
Bye.
